### PR TITLE
chore: upgrade AWS kotlin SDK

### PR DESCRIPTION
**Pull request by div5yesh**
- [x] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*
- Upgrade Kotlin SDK version
- add new imports

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by gpanshu — APPROVED** [review]

File: aws-auth-cognito/build.gradle [hunk 1]
```diff
@@ -34,6 +34,7 @@ dependencies {
     implementation dependency.aws.cognitoidentityprovider
 
     testImplementation project(path: ':testutils')
+    testImplementation project(path: ':aws-auth-cognito')
```
Comment (sdhuka) [inline aws-auth-cognito/build.gradle:37]: why is this required?

File: build.gradle [hunk 2]
```diff
@@ -141,7 +141,7 @@ ext {
         mockito: 'org.mockito:mockito-core:3.9.0',
         mockk: 'io.mockk:mockk:1.12.3',
         mockkandroid: 'io.mockk:mockk-android:1.12.3',
-        mockwebserver: 'com.squareup.okhttp3:mockwebserver:4.9.1',
+        mockwebserver: 'com.squareup.okhttp3:mockwebserver:5.0.0-alpha.9',
```
Comment (sdhuka) [inline build.gradle:144]: why are we using external alpha release?

File: aws-auth-cognito/build.gradle (hunk 1 above)
Comment (div5yesh) [reply aws-auth-cognito/build.gradle:37]: AWS Kotlin SDK upgraded the Kotlin version to 1.7.10 and it was resulting in build errors. So we had to upgrade the Kotlin version too. New version requires explicit main compilation dependency on test implementation. Not sure if there is a better way to do this. Please advice.

File: build.gradle (hunk 2 above)
Comment (div5yesh) [reply build.gradle:144]: AWS Kotlin SDK is using alpha release which is causing build failures due to dependency version conflicts.

**Review by sdhuka — APPROVED** [review]
